lib/shim/message-shim.js

```diff
@@ -438,16 +440,18 @@ function recordConsume(nodule, properties, spec) {
       // Call the method in the context of our segment.
       let ret = shim.applySegment(fn, segment, true, this, args)
 
-      // Intercept the promise to handle the result.
-      if (resHandler && ret && msgDesc.promise && shim.isPromise(ret)) {
+      if (ret && shim.isPromise(ret)) {
         ret = shim.bindPromise(ret, segment)
-        ret = ret.then(function interceptValue(res) {
-          const msgProps = resHandler.call(this, shim, fn, fnName, res)
-          if (getParams && msgProps && msgProps.parameters) {
-            shim.copySegmentParameters(segment, msgProps.parameters)
-          }
-          return res
-        })
+        // Intercept the promise to handle the result.
+        if (resHandler && msgDesc.promise) {
```

Comment:
Minor but I think the spec `.promise` is used as a check on if the instrumentation intends to be doing promise stuff. Maybe as a check first to avoid doing the `typeof` (via `shim.isPromise()`) on every returned result? Anyways, I'd prob have the `msgDesc.promise` check above prior to `shim.isPromise()` and remove from down here.

Part of me wonders why we have these checks and don't just determine based on return value but perhaps there are cases where a promise is returned but we've handled slightly differently that works? 🤷. Something to dig into at some point.

Reply:
this whole code path gives me 🤷🏻 responses.  I'll just hoist this .promise check to the first if on line 443 for now.  I did log https://github.com/newrelic/node-newrelic/issues/981 to do more research into this